src/query/service/src/pipelines/processors/transforms/transform_add_const_columns.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_catalog::table_context::TableContext;
+use common_exception::Result;
+use common_expression::DataBlock;
+use common_expression::DataSchemaRef;
+use common_expression::Expr;
+use common_expression::Scalar as DataScalar;
+use common_sql::evaluator::BlockOperator;
+use common_sql::evaluator::CompoundBlockOperator;
+
+use crate::pipelines::processors::port::InputPort;
+use crate::pipelines::processors::port::OutputPort;
+use crate::pipelines::processors::processor::ProcessorPtr;
+use crate::pipelines::processors::transforms::transform::Transform;
+use crate::pipelines::processors::transforms::transform::Transformer;
+use crate::sessions::QueryContext;
+
+pub struct TransformAddConstColumns {
+    expression_transform: CompoundBlockOperator,
+    input_len: usize,
+}
+
+impl TransformAddConstColumns
+where Self: Transform
+{
+    pub fn try_create(
+        input: Arc<InputPort>,
+        output: Arc<OutputPort>,
+        input_schema: DataSchemaRef,
+        output_schema: DataSchemaRef,
+        mut const_values: Vec<DataScalar>,
+        ctx: Arc<QueryContext>,
```

Comment:
It would be better to move `ctx: Arc<QueryContext>` to the first parameter.

Reply:
what about make it the 3rd?
input, output are always needed for a Transform?

Reply:
fixed